Sganes21 classifier

sganes21 opened the pull request:
- Modified svm to take in square pixel images
- created load image function to take in square images from 32x32 to 256x 256
- created svm_model train function to take in a set of image info and associated labels to train and generate an svm model
- created a svm prediction function to take in new image data and associated svm model and output a prediction
- integrated the load image and svm model train function within app.py
- modified requirements.txt file

sganes21 committed:
created construct for svm classifier

sganes21 committed:
added changes to classifier to all to read labels off filename

sganes21 committed:
svm incorporation to gradio app

auto-d commented on app/app.py:
```diff
@@ -227,7 +229,33 @@ def generate_images(size_pixels, angle_increment):
     annotations.to_csv(get_val_csv(), index=False)
 
-    return examples
+    image_data_list = []
```
This function creates our test images from the 3d models. It's used to help construct the dataset we use to train the classifiers. I think it is not where we want to train the SVM? Refactoring the `train_model` function below would make a better fit perhaps.

auto-d commented on app/app.py:
```diff
@@ -227,7 +229,33 @@ def generate_images(size_pixels, angle_increment):
+        label = row[2]
+
+    for x in X:
+            for y in Y:
```
X, Y, Z are not defined in this scope. Is this a copy/paste error?

auto-d commented on app/app.py:
```diff
@@ -227,7 +229,33 @@ def generate_images(size_pixels, angle_increment):
+
+
+
+    return examples, svm_model
```
Changing this return type to a tuple will break the calling function which expects an array of images.

sganes21 committed:
update svm code and test

sganes21 commented:
- Update:

- Removed the integration of svm into app.py as it was crashing the application
- updated the svm with the following functions:
1) load_image(path)
updated load image function to take in square images from 32x32 to 500x500

2) load_dataset
takes in the data directory and associated category labels from subdirectory folders
returns a numpy array of the data, labels, and associated filenames
added error handling in case the loading fails

3) train_validate_model
takes in the data directory, category labels, test size=0.2 and random state=0
optionally allows to save model for debugging purposes
using accuracy to evaluate model (published using scikit learn classification report)
scaling using StandardScaler
stratify category labels as best practice

4) predict_on_new_image
updated to take in model, scaler, categories, and image paths
calculates predictions but also added as output the test image with the prediction. 


- updated experimentation.ipynb with the following:

1) Setting up the Data Directories

takes in the stl filename which contains the label (i.e. "C1"), creates a data subdirectory with that label, and outputs the filename as png with lighting and specified rotations. The unique patient identifier is included in the filename output to avoid unintended overwriting. 

2) Setting up SVM

Shows how to use svm.py. 
PLEASE NOTE: Do not put Test folder containing the image(s) for generating predictions in the data directory as doing so will result in this image to be included in the training and validating of the SVM model.

auto-d committed:
merged main

auto-d committed:
superficial changes to NN training

auto-d committed:
integrated svm w/ app

auto-d reviewed (commented):
Refactoring resolves issues. Tks!
